core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -54,7 +54,7 @@ import org.apache.spark.util.{AccumulatorV2, ThreadUtils, Utils}
 private[spark] class TaskSchedulerImpl private[scheduler](
     val sc: SparkContext,
     val maxTaskFailures: Int,
-    private[scheduler] val blacklistTrackerOpt: Option[BlacklistTracker],
+    mockBlacklistTracker: Option[BlacklistTracker] = None,
```

Comment:
@squito the previous code here in `maybeCreateBlacklistTracker` has initialing ordering issue, this method is called before `schedulerBackend` is created, so the `ExecutorAllocationClient` we got will always be None. So I'm trying to address this issue. Let me find out a better way to handle it.